File: coral-model/src/main/kotlin/io/coral/contacts/model/repository/BasicRepository.java
package io.coral.contacts.model.repository;

import io.coral.contacts.model.domain.AbstractEntity;
import java.util.List;

public interface BasicRepository {

      AbstractEntity add(AbstractEntity entity);

      AbstractEntity update(AbstractEntity entity);

      AbstractEntity delete(AbstractEntity entity);

      List<AbstractEntity> findAll(String entity, int offset, int maxSize);

}

File: coral-model/src/main/kotlin/io/coral/contacts/model/repository/BasicRepository.kt
package io.coral.contacts.model.repository


import io.tech4health.ts.model.domain.AbstractEntity

interface BasicRepository {

    fun doAdd(entity: AbstractEntity): AbstractEntity

    fun doUpdate(entity: AbstractEntity): AbstractEntity

    fun doDelete(entity:AbstractEntity,id:Int): Boolean?

    fun findAll(entity: String, offset: Int, maxSize: Int): List<AbstractEntity>

}
